code/modules/mapping/map_template.dm

```diff
@@ -71,31 +103,67 @@
 	parsed.initTemplateBounds()
 	smooth_zlevel(world.maxz)
 	log_game("Z-level [name] loaded at [x],[y],[world.maxz]")
+	on_map_loaded(world.maxz, parsed.bounds)
 
 	return level
 
-/datum/map_template/proc/load(turf/T, centered = FALSE)
+//Override for custom behavior
+/datum/map_template/proc/on_map_loaded(z, list/bounds)
+	loaded++
+
+/**
+  * Proc to trigger a load at a specific area. Calls on_map_loaded(T.z, loaded_bounds) afterwards.
+  *
+  * @params
+  * * turf/T - Turf to load at
+  * * centered - Center at T or load with the bottomright corner being at T
+  * * orientation - SOUTH is default, anything else rotates the map to face it with the point of reference being the map itself is facing south by default. Cardinals only, don't be a 4head and put in multiple flags. It won't work or be pretty if you try.
+  * * annihilate - Should we destroy stuff in our bounds while loading
+  * * force_cache - Should we force the parsed shuttle to cache instead of being GC'd post loading if it wasn't going to be cached by default
+  * * rotate_placement_to_orientation - Has no effect if centered. Should we rotate where we load it around the turf we're loading at? Used for stuff like engine submaps when the station is rotated.
+  *
+  */
+/datum/map_template/proc/load(turf/T, centered = FALSE, orientation = SOUTH, annihilate = default_annihilate, force_cache = FALSE, rotate_placement_to_orientation = FALSE)
+	var/old_T = T
 	if(centered)
-		T = locate(T.x - round(width/2) , T.y - round(height/2) , T.z)
+		T = locate(T.x - FLOOR(((orientation & (NORTH|SOUTH))? width : height) / 2, 1) , T.y - FLOOR(((orientation & (NORTH|SOUTH)) ? height : width) / 2, 1) , T.z) // %180 catches East/West (90,270) rotations on true, North/South (0,180) rotations on false
```

Comment:
```suggestion
		T = locate(T.x - round(((orientation & (NORTH|SOUTH))? width : height) / 2) , T.y - round(((orientation & (NORTH|SOUTH)) ? height : width) / 2) , T.z) // %180 catches East/West (90,270) rotations on true, North/South (0,180) rotations on false
```
Feel free to disregard these suggestions if the FLOOR macro turns out to be more efficient than the integrated round() function.